src/main/java/net/dries007/tfc/util/InteractionManager.java

```diff
@@ -234,6 +237,34 @@ public static void setup()
             return ActionResultType.PASS;
         });
 
+        register(TFCTags.Items.SCRAPABLE_ITEM, (stack, context) -> {
+            World level = context.getLevel();
+            ScrapingRecipe recipe = ScrapingRecipe.getRecipe(level, new ItemStackRecipeWrapper(stack));
+            if (recipe != null)
+            {
+                BlockPos pos = context.getClickedPos();
+                BlockPos abovePos = pos.above();
+                PlayerEntity player = context.getPlayer();
+                if (player != null && context.getClickedFace() == Direction.UP && level.getBlockState(pos).is(TFCTags.Blocks.SCRAPING_SURFACE) && level.getBlockState(abovePos).isAir())
+                {
+                    level.setBlockAndUpdate(abovePos, TFCBlocks.SCRAPING.get().defaultBlockState());
+                    final ScrapingTileEntity te = Helpers.getTileEntity(level, abovePos, ScrapingTileEntity.class);
+                    if (te != null)
+                    {
+                        return Helpers.getCapability(te, CapabilityItemHandler.ITEM_HANDLER_CAPABILITY).map(cap -> {
+                            if (!level.isClientSide)
+                            {
+                                player.setItemInHand(context.getHand(), cap.insertItem(0, stack, false));
+                                te.cachedItem = recipe.getResultItem().copy();
```

Comment:
This should use `assemble`, and not perform a copy() (because `assemble` does.) `getResultItem` is for display purposes (generally). the same wrapper can be re-used from earlier in this method.

Reply:
cachedItem is specifically for display purposes only. assemble is called when the TE realizes it's complete